codesign/test/codesign_test.dart

```diff
@@ -2,11 +2,32 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'package:codesign/codesign.dart';
 import 'package:test/test.dart';
+import 'package:codesign/codesign.dart' as cs;
+
+class FakeCodesignContext extends cs.CodesignContext {
+  FakeCodesignContext(
+      {required super.codesignCertName,
+      required super.codesignUserName,
+      required super.appSpecificPassword,
+      required super.codesignAppstoreId,
+      required super.codesignTeamId,
+      required super.commitHash,
+      super.production = false});
+}
 
 void main() {
-  test('calculate', () {
-    expect(calculate(), 42);
+  const String randomString = 'abcd1234';
+
+  FakeCodesignContext codesignContext = FakeCodesignContext(
+      codesignCertName: randomString,
+      codesignUserName: randomString,
+      appSpecificPassword: randomString,
+      codesignAppstoreId: randomString,
+      codesignTeamId: randomString,
+      commitHash: randomString);
+
+  test('nothing', () async {
```

Comment:
Implementing a basic process execution on each codesign visitor would make this testable. This would let you ensure the correct nodes are processed.